ofonotextchannel.cpp

```diff
@@ -480,13 +480,31 @@ void oFonoTextChannel::mmsReceived(const QString &id, uint handle, const QVarian
     header["message-token"] = QDBusVariant(id);
     header["message-sender"] = QDBusVariant(handle);
     header["message-received"] = QDBusVariant(QDateTime::currentDateTimeUtc().toTime_t());
+    if (properties["Received"].isNull() == false && properties["Received"].toUInt() > 0) {
```

Comment:
Can you do the "if" first, and then do current Date and time in the else part? so we do not execute code thats unnecessary

Reply:
Done.